torch/utils/_benchmark/utils/valgrind_wrapper.py

```diff
@@ -0,0 +1,227 @@
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+
+import torch
+
+
+class _ValgrindWrapper(object):
+    def __init__(self):
+        self._valgrind_available = not subprocess.run(
+            ["which", "valgrind"], capture_output=True).returncode
+
+        self._build_type = None
+        build_search = re.search("BUILD_TYPE=(.+),", torch.__config__.show())
+        if build_search is not None:
+            self._build_type = build_search.groups()[0].split(",")[0]
+
+        self._baseline_cache = {}
+
+    def _validate(self):
+        if not self._valgrind_available:
+            raise OSError("Could not find `valgrind`")
+
+        if self._build_type is None:
+            raise OSError("Could not determine BUILD_TYPE")
+
+        if self._build_type != "RelWithDebInfo":
+            raise OSError(
+                "PyTorch sould be built with REL_WITH_DEB_INFO=1 "
+                "(debug symbols) in order to produce a useful C++ call graph, "
+                f"got BUILD_TYPE={self._build_type} instead"
```

Comment:
You can still get instructions retired though?

Reply:
Yeah, I actually didn't realize how much information was preserved even with `Release`. I've switched to a simple warning in the `__repr__` for the output object advising that `REL_WITH_DEB_INFO=1` could yield more info, but definitely agree that it doesn't make sense to refuse to collect unless we have debug symbols.